### PR TITLE
[ISSUE #1147]Add name server domain host support

### PR DESCRIPTION
**Pull request by duhenglucky**
## What is the purpose of the change

Add nameserver host support

## Brief changelog

XX

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 5 checklist(the last one is not necessary)before request the community to review your PR`.

- [x] Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/rocketmq/tree/master/test).
- [x] Run `mvn -B clean apache-rat:check findbugs:findbugs checkstyle:checkstyle` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.
- [x] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

File: client/src/main/java/org/apache/rocketmq/client/ClientConfig.java [hunk 1]
```diff
@@ -241,6 +247,15 @@ public void setLanguage(LanguageCode language) {
     }
```
Comment (zongtanghu) [inline client/src/main/java/org/apache/rocketmq/client/ClientConfig.java]: Here,you should add some code annotation.

File: client/src/main/java/org/apache/rocketmq/client/ClientConfig.java [hunk 2]
```diff
@@ -241,6 +247,15 @@ public void setLanguage(LanguageCode language) {
...
 
     public String getNamespace() {
+        if (StringUtils.isNotEmpty(namespace)) {
+            return namespace;
+        }
+
+        if (StringUtils.isNotEmpty(this.namesrvAddr)) {
```
Comment (zongtanghu) [inline client/src/main/java/org/apache/rocketmq/client/ClientConfig.java]: Here,do you mean that if namespace is empty,then return the value by parseInstanceIdFromEndpoint() method?

File: client/src/main/java/org/apache/rocketmq/client/ClientConfig.java [hunk 3]
```diff
@@ -241,6 +247,15 @@ public void setLanguage(LanguageCode language) {
...
+            if (NameServerAddressUtils.validateInstanceEndpoint(namesrvAddr)) {
+                return NameServerAddressUtils.parseInstanceIdFromEndpoint(namesrvAddr);
+            }
+        }
         return namespace;
```
Comment (zongtanghu) [inline client/src/main/java/org/apache/rocketmq/client/ClientConfig.java]: Actually,I think the last code('return namespace') will return null!I think,so you can return null directly!

**Comment by zongtanghu** [comment]
@duhenglucky  I add some code review comments.

File: client/src/main/java/org/apache/rocketmq/client/ClientConfig.java (hunk 2 above)
Comment (duhenglucky) [reply client/src/main/java/org/apache/rocketmq/client/ClientConfig.java]: yep，we can get the namespace by parse Access point UR

File: client/src/main/java/org/apache/rocketmq/client/ClientConfig.java (hunk 3 above)
Comment (duhenglucky) [reply client/src/main/java/org/apache/rocketmq/client/ClientConfig.java]: just keep insistent with the original get method.

**Pull request merged**